build.sbt

```
@@ -1,24 +1,55 @@
-import uk.gov.hmrc.DefaultBuildSettings
-
-ThisBuild / majorVersion := 0
-ThisBuild / scalaVersion := "2.13.12"
+import com.typesafe.sbt.web.PathMapping
+import com.typesafe.sbt.web.pipeline.Pipeline
+import play.sbt.PlayImport.PlayKeys.playDefaultPort
+import uk.gov.hmrc.DefaultBuildSettings._
+import scoverage.ScoverageKeys
+import uk.gov.hmrc.sbtdistributables.SbtDistributablesPlugin.publishingSettings
 
 lazy val microservice = Project("pillar2-submission-api", file("."))
   .enablePlugins(play.sbt.PlayScala, SbtDistributablesPlugin)
   .settings(
+    majorVersion := 0,
+    ScoverageKeys.coverageExcludedFiles :=
+      "<empty>;com.kenshoo.play.metrics.*;.*definition.*;prod.*;testOnlyDoNotUseInAppConf.*;" +
+        "app.*;.*BuildInfo.*;.*Routes.*;.*repositories.*;.*controllers.platform.*;.*controllers.test.*;.*services.test.*;.*metrics.*",
+    ScoverageKeys.coverageMinimumStmtTotal := 80,
+    ScoverageKeys.coverageFailOnMinimum := true,
+    ScoverageKeys.coverageHighlighting := true,
+    Compile / scalafmtOnCompile      := true,
+    Test / scalafmtOnCompile         := true,
+    playDefaultPort := 10054,
     libraryDependencies ++= AppDependencies.compile ++ AppDependencies.test,
     // https://www.scala-lang.org/2021/01/12/configuring-and-suppressing-warnings.html
     // suppress warnings in generated routes files
-    scalacOptions += "-Wconf:src=routes/.*:s",
+    scalacOptions += "-Wconf:src=routes/.*:s"
```

We should also add more Scala options to enforce conditions like dead code etc 

```suggestion
    scalacOptions ++= Seq(
       "-Wconf:src=routes/.*:s",
       "-Ywarn-dead-code",
      "-Xfatal-warnings",
      "-Ywarn-unused:implicits", // Warn if an implicit parameter is unused.
      "-Ywarn-unused:imports",   // Warn if an import selector is not referenced.
      "-Ywarn-unused:locals",    // Warn if a local definition is unused.
      "-Ywarn-unused:params",    // Warn if a value parameter is unused.
      "-Ywarn-unused:patvars",   // Warn if a variable bound in a pattern is unused.
      "-Ywarn-unused:privates"   // Warn if a private member is unused.
    ),
    
```

---

Thanks Tevyn. I had to remove -Xfatal-warnings just to avoid errors, we can add later once start coding.